predict-daily-temperature/Temperature-Demo.ipynb

```diff
@@ -0,0 +1,6470 @@
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 22,
+   "id": "9c92cff9",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "def remove_nans(time_target_fs, target_col):\n",
+    "    # remove nans\n",
+    "    max_nans = 0\n",
+    "    for col in time_target_fs.columns:\n",
+    "        max_nans = max(time_target_fs[col].isna().sum(), max_nans)\n",
+    "\n",
+    "    if max_nans:\n",
+    "        return time_target_fs.iloc[max_nans:]\n",
+    "    \n",
+    "    return time_target_fs\n",
+    "\n",
+    "# Get rid of the time index column for modeling\n",
+    "baseline_training.drop(time_index, axis=1, inplace=True)\n",
+    "baseline_test.drop(time_index, axis=1, inplace=True)\n",
+    "\n",
+    "# The lag feature introduces nans, so we remove those rows and pull out the target\n",
+    "X_train = remove_nans(baseline_training, target_col)\n",
+    "y_train = X_train.pop(target_col)\n",
+    "\n",
+    "X_test = remove_nans(baseline_test, target_col)\n",
+    "y_test = X_test.pop(target_col)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "e18cc5f9",
+   "metadata": {},
+   "source": [
+    "### Build The Model"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 23,
+   "id": "412e13e9",
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "Median Abs Error: 1.99\n",
+      "1: target_lag [1.000]\n",
+      "-----\n",
+      "\n"
+     ]
+    }
+   ],
+   "source": [
+    "reg = RandomForestRegressor(n_estimators=100)\n",
+    "reg.fit(X_train, y_train)\n",
+    "\n",
+    "preds = reg.predict(X_test)\n",
+    "scores = median_absolute_error(preds, y_test)\n",
+    "print('Median Abs Error: {:.2f}'.format(scores))\n",
+    "\n",
+    "high_imp_feats = feature_importances(X_train, reg, feats=10)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "ee8d9a32",
+   "metadata": {},
+   "source": [
+    "Next, we're going to expand our feature engineering step using Featuretools' deep feature synthesis. We'll utilize a few time-series specific primitives as well as some datetime primitives. \n",
+    "\n",
+    "# Feature Engineering Run \n",
+    "\n",
+    "We're going to use the same split that we used for the baseline run, so we'll work with the `training_data` and `test_data` dataframes from before."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "04b32e54",
+   "metadata": {},
+   "source": [
+    "### Feature Engineering with Featuretools\n",
+    "\n",
+    "In order to use DFS, we need to set up `EntitySet` objects for both the training and testing data. Since our dataset only includes one table, this will be a quick process. "
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 24,
+   "id": "995348c8",
+   "metadata": {},
+   "outputs": [
+    {
+     "data": {
+      "text/plain": [
+       "Entityset: None\n",
+       "  DataFrames:\n",
+       "    temperatures [Rows: 1169, Columns: 3]\n",
```

Comment:
Maybe add an inline comment about how the index is one of the columns, since users would probably be expecting 2 columns from the earlier data?